controllers/login.js

```diff
@@ -484,4 +484,27 @@ router.get('/logout/', (req, res, next) => {
 		.catch(next);
 });
 
+router.get('/logout/external/', async (req, res, next) => {
+	let redirectUri = '/logout/';
+	if (Configuration.has('OAUTH2_LOGOUT_URI')) {
+		redirectUri = Configuration.get('OAUTH2_LOGOUT_URI');
+	}
+
+	if (res.locals.isExternalLogoutAllowed) {
+		try {
+			await api(req, { version: 'v3' }).post('/logout/external');
+		} catch (err) {
+			logger.error('error during external logout.', formatError(err));
+			req.session.notification = {
+				type: 'danger',
+				message: res.$t('logout.text.externalLogoutFailed', { systemName: res.locals.systemName ?? '' }),
+			};
+			res.redirect('/dashboard');
+		}
+	}
+
+	res.statusCode = 307;
```

Comment:
I don't think you need to set the status code manually. This is usually done by `res.redirect` automatically. Otherwise `res.redirect(307, redirectUri)` would be the correct method to overwrite the status code.